altapay.php

```diff
@@ -0,0 +1,2353 @@
+		`cardExpiryDate` varchar(200) DEFAULT '' NOT NULL,
+		`ccToken` varchar(200) DEFAULT '' NOT NULL,
+		PRIMARY KEY  (`id`)
+		) ENGINE=" . _MYSQL_ENGINE_ . "  DEFAULT CHARSET=utf8 AUTO_INCREMENT=1");
+        }
+
+
+        if (!Db::getInstance()->Execute("ALTER TABLE `" . _DB_PREFIX_ . "altapay_orderlines`
+            MODIFY `product_id` varchar(36) NOT NULL")) {
+            $this->context->controller->errors[] = Db::getInstance()->getMsgError();
+            return false;
+        }
+
+        /* This table captures the payment information */
+        if (Db::getInstance()->Execute("SELECT 1 FROM `" . _DB_PREFIX_ . "valitor_cartInfo`")) {
+
+            $sql = "RENAME TABLE  `" . _DB_PREFIX_ . "valitor_cartInfo`  TO `" . _DB_PREFIX_ . "altapay_cartInfo`  ";
+            Db::getInstance()->Execute($sql);
+
+        } else {
+            Db::getInstance()->Execute('CREATE TABLE IF NOT EXISTS `' . _DB_PREFIX_ . 'altapay_cartInfo` (
+			`id_cart` int(10) unsigned NOT NULL,
+			`productDetails` varchar(255) NOT NULL,
+			`date_add` varchar(50) NOT NULL,
+			PRIMARY KEY (`id_cart`)
+		) ENGINE=' . _MYSQL_ENGINE_ . '  DEFAULT CHARSET=utf8');
+        }
+
+
+        $this->createOrderState();
+
+        return true;
+    }
+
+    /**
+     * Create a new order state
+     */
+    public function createOrderState()
+    {
+        if (!Configuration::get('ALTAPAY_OS_PENDING')) {
+            $orderState = new OrderState();
+            $orderState->name = array();
+
+            foreach (Language::getLanguages() as $language) {
+                $orderState->name[$language['id_lang']] = 'Awaiting payment processing';
+            }
+
+            $orderState->color = '#ffff5a';
+
+            $orderState->logable = false;
+            $orderState->invoice = false;
+            $orderState->hidden = false;
+            $orderState->send_email = false;
+            $orderState->shipped = false;
+            $orderState->paid = false;
+            $orderState->delivery = false;
+
+            if ($orderState->add()) {
+                $source = dirname(__FILE__) . '/views/img/os_pending.gif';
+                $destination = dirname(__FILE__) . '/../../img/os/' . (int)$orderState->id . '.gif';
+                copy($source, $destination);
+            }
+            Configuration::updateValue('ALTAPAY_OS_PENDING', (int)$orderState->id);
+        }
+    }
+
+    /**
+     * Called on uninstall
+     * Leaves tables in place in order to not loose history.
+     */
+    public function uninstall()
+    {
+        if (!Configuration::deleteByName('ALTAPAY_USERNAME')
+            || !Configuration::deleteByName('ALTAPAY_PASSWORD')
+            || !Configuration::deleteByName('ALTAPAY_URL')
+            || !parent::uninstall()
+        ) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Return content for the configuration in back office
+     *
+     * @return String HTML for display
```

Comment:
string is not a class in PHP
```suggestion
     * @return string HTML for display
```